### PR TITLE
Almost Happy Linter

### PR DESCRIPTION
**Pull request by kensipe**
Signed-off-by: Ken Sipe <kensipe@gmail.com>

**Description of the change:**
Updates based on issues identified with `make lint`.  It was large and this is PR 1.  It is largely not controversial changes IMO.   This PR reduces linter issues from 38 to 18.

`make lint | grep -o ':[[:digit:]].*' | wc -l`

What is remaining requires a conversation.  the following (remaining) I think will be easy:

* write file permissions (gosec)
* subprocess launch (gosec)
* ignored error returns (errcheck)

The remaining more challenging is:
* cyclomatic complexity (gocyclo) *3
* weak and blacklisted crypto for md5 (gosec)
* decompression bomb (gosec)

**Motivation for the change:**

visions of clean and shiny code

**Checklist**

If the pull request includes user-facing changes, extra documentation is required:
- [ ] Add a new changelog fragment in `changelog/fragments` (see [`changelog/fragments/00-template.yaml`](https://github.com/operator-framework/operator-sdk/tree/master/changelog/fragments/00-template.yaml))
- [ ] Add or update relevant sections of the docs website in [`website/content/en/docs`](https://github.com/operator-framework/operator-sdk/tree/master/website/content/en/docs)

### CODE REVIEW

**Comment by kensipe** [comment]
rebased
cc: @joelanford @estroz

File: hack/generate/changelog/util/changelog_test.go [hunk 1]
```diff
@@ -283,6 +283,7 @@ No changes for this release!
 	}
 
 	for _, tc := range testCases {
+		tc := tc
```
Comment (jmrodri) [inline hack/generate/changelog/util/changelog_test.go:286]: What linter error is this fixing? I find it strange to make a copy of the variable in the loop. I've never had to do this.

File: images/scorecard-test-kuttl/cmd/test-kuttl/main.go [hunk 2]
```diff
@@ -49,7 +49,7 @@ func main() {
 	}
 
 	var jsonReport Testsuites
-	err = json.Unmarshal([]byte(byteValue), &jsonReport)
+	err = json.Unmarshal(byteValue, &jsonReport)
```
Comment (jmrodri) [inline images/scorecard-test-kuttl/cmd/test-kuttl/main.go:52]: +1 to removing the `[]byte(..)`.

File: internal/ansible/controller/status/types.go [hunk 3]
```diff
@@ -77,7 +77,10 @@ func NewAnsibleResultFromMap(sm map[string]interface{}) *AnsibleResult {
 		a.Failures = int(v.(int64))
 	}
 	if v, ok := sm["completion"]; ok {
-		s := v.(string)
+		s, ok := v.(string)
+		if !ok {
+			log.V(0).Info("Unexpected non-string value for `completion` key")
+		}
```
Comment (jmrodri) [inline internal/ansible/controller/status/types.go:83]: This makes sense. Probably shouldn't ignore errors.

File: internal/generate/clusterserviceversion/bases/definitions/ast.go [hunk 4]
```diff
@@ -31,6 +31,7 @@ func (g generator) getMarkedChildrenOfField(root markers.FieldInfo) (map[string]
 	for len(nextFields) > 0 {
 		fields := []*fieldInfo{}
 		for _, field := range nextFields {
+			field := field
```
Comment (jmrodri) [inline internal/generate/clusterserviceversion/bases/definitions/ast.go:34]: -1 to reassigning to a local variable in the loop

File: internal/olm/operator/internal/registry_pod.go [hunk 5]
```diff
@@ -185,8 +185,7 @@ func (rp *RegistryPod) VerifyPodRunning(ctx context.Context) error {
 // checkPodStatus polls and verifies that the pod status is running
 func (rp *RegistryPod) checkPodStatus(ctx context.Context, podCheck wait.ConditionFunc) error {
 	// poll every 200 ms until podCheck is true or context is done
-	err := wait.PollImmediateUntil(time.Duration(200*time.Millisecond),
-		podCheck, ctx.Done())
+	err := wait.PollImmediateUntil(200*time.Millisecond, podCheck, ctx.Done())
```
Comment (jmrodri) [inline internal/olm/operator/internal/registry_pod.go:188]: +1 to the simplification

File: internal/generate/clusterserviceversion/bases/definitions/ast.go (hunk 4 above)
Comment (kensipe) [reply internal/generate/clusterserviceversion/bases/definitions/ast.go:34]: @jmrodri this is not reassignment of a local var...   This is resolving "Using reference to loop iterator variable" as identified as mistake 1 in the [CommonMistakes wiki](https://github.com/golang/go/wiki/CommonMistakes#using-reference-to-loop-iterator-variable).

**Comment by kensipe** [comment]
@jmrodri 
```
golangci-lint has version 1.29.0 built
```

File: internal/generate/clusterserviceversion/bases/definitions/ast.go (hunk 4 above)
Comment (kensipe) [reply internal/generate/clusterserviceversion/bases/definitions/ast.go:34]: technically it is the creation of a new variable with the same name assigned to the value of the original "field", but allows for the dereferencing of an address to the correct value which the original "field" does not.

Comment (kensipe) [reply internal/generate/clusterserviceversion/bases/definitions/ast.go:34]: without this fix... 
```
internal/generate/clusterserviceversion/bases/definitions/ast.go:71:29: Using the variable on range scope `field` in function literal (scopelint)
							pathSegments: append(field.pathSegments, segment),
```

**Comment by kensipe** [comment]
@estroz @jmrodri please have another look... I've reduced to bare min... and while the current linters don't caught the issue at line 244 of olm.go, it is a bug which is resolved by that line change.

**Comment by kensipe** [comment]
expect a different PR (but I'll be back :) )

**Pull request closed without merging**